Source/Particles/ParticleBoundaryBuffer.cpp

```diff
@@ -159,6 +160,9 @@ void ParticleBoundaryBuffer::gatherParticles (MultiParticleContainer& mypc,
     const amrex::Geometry& geom = warpx_instance.Geom(0);
     auto plo = geom.ProbLoArray();
     auto phi = geom.ProbHiArray();
+
+    int *pre_count = static_cast<int*>(amrex::The_Managed_Arena()->alloc(sizeof(size_t)));
```

Comment:
I suspect it will perform better to use `amrex::ReduceOps` for this, rather than atomically incrementing a managed memory counter, as here for example: 

https://github.com/ECP-WarpX/WarpX/blob/development/Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp#L451

More worried about systems like Crusher where the managed memory performance is still pretty bad than NVIDIA...

Reply:
Yeah, this was an admittedly lazy, expedient approach. Managed memory is a problem. Revised to use the reduction approach, and testing now.